### DISCUSSION TITLE
Version Question: Why the latest version of this extension downloaded from visual studio market is 1.9.1?

### DISCUSSION THREAD

**Discussion by inlann** [microsoft/vscode-cpptools#8953] [Q&A] [answered]
Hello:

May I ask why the latest version of cpptools downloaded from [visual studio marketplace](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) is 1.9.1 but the latest version here is still 1.8.4?

And the [change log](https://marketplace.visualstudio.com/items/ms-vscode.cpptools/changelog) in marketplace is newer than [github](https://github.com/microsoft/vscode-cpptools/blob/release/Extension/CHANGELOG.md)

Since I modified the source code of cpptools release branch and the version is 1.8.2-main and I need to download the binaries `cpptools` and `cpptools-srv`. Should I copy them from the latest cpptools downloaded from marketplace?

**Comment by sean-mcmanus** [comment, marked as answer]
1.9.1 is a "pre-release" (insiders) version (updates faster, up to once a week or two). 1.8.4 is the "release" version (updates slower, every 6 weeks or so). The latest pre-release version may have a higher chance of having a regression (e.g. currently there is 1 known crash regression that will be fixed in 1.9.2).

The release branch on GitHub matches binaries from 1.8.4 (the insiders branch matches 1.9.1 and main matches our pending 1.9.2). The version in the package.json is typically for the last previous release, which is why it shows 1.8.2-main.

**Reply by inlann** [reply]
That's really helpful! Thanks 😄
